app/src/main/java/com/keylesspalace/tusky/BaseActivity.java

```diff
@@ -222,4 +235,37 @@ public void showAccountChooserDialog(CharSequence dialogTitle, boolean showActiv
             .setAdapter(adapter, (dialogInterface, index) -> listener.onAccountSelected(accounts.get(index)))
             .show();
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode != REQUESTER_NONE && requesters.containsKey(requestCode)) {
+            requesters.remove(requestCode).onRequestPermissionsResult(permissions, grantResults);
+        }
+    }
+
+    public boolean requestPermissions(String[] permissions, int minimumBuildVersion, @Nullable PermissionRequester requester) {
+        if (minimumBuildVersion == BUILD_VERSION_ANY || Build.VERSION.SDK_INT >= minimumBuildVersion) {
+            ArrayList<String> permissionsToRequest = new ArrayList<>();
+            for(String permission: permissions) {
+                if (ContextCompat.checkSelfPermission(this, permission) != PackageManager.PERMISSION_GRANTED) {
+                    permissionsToRequest.add(permission);
+                }
+            }
+            if (permissionsToRequest.isEmpty()) {
+                return true;
+            }
+
+            int newKey = requester == null ? REQUESTER_NONE : requesters.size();
+            String[] permissionsCopy = new String[permissionsToRequest.size()];
+            permissionsToRequest.toArray(permissionsCopy);
+            ActivityCompat.requestPermissions(this, permissionsCopy, newKey);
+            if (newKey != REQUESTER_NONE) {
```

Comment:
I would put request key there before requesting the permission. I had such bugs before when permissions are given right away or similar.